src/ring_theory/polynomial/chebyshev/dickson.lean

```diff
@@ -36,84 +37,25 @@ noncomputable theory
 
 namespace polynomial
 
-variables {R S : Type*} [comm_ring R] [comm_ring S]
-
-/-- `dickson'` is an auxiliary non-standard function that lets us unify some proofs for `dickson₁`
-and `dickson₂`.
--/
-noncomputable def dickson' (p₀ : polynomial R) (a : R) : ℕ → polynomial R
-| 0       := p₀
-| 1       := X
-| (n + 2) := X * dickson' (n + 1) - (monomial 0 a) * dickson' n
```

Comment:
Did you mean to revert this design? The commit message suggests you didn't

Reply:
I was experimenting with that. I had the impression that this just led to many more lemmas that no one would use because these are not studied in the literature without having the benefit of making things easier. The new generalised dickson function actually felt like an improvement as it allowed me to take the first and second as an additional argument. What is still open is to redefine it completely in terms of `linear_recurrence' which will certainly be useful later on (e.g. when proving explicit formulae for the dickson polynomials).